tests/e2e/test_execute_async.py

```diff
@@ -53,10 +83,112 @@ def test_cant_get_results_after_cancel(self, long_running_ae: AsyncExecution):
+        query_id, query_secret = str(long_running_ae.query_id), str(
+            long_running_ae.query_secret
+        )
+
+        def cancel_query_in_separate_thread(query_id, query_secret):
+            with self.connection() as conn:
+                ae = conn.get_async_execution(query_id, query_secret)
+                ae.cancel()
+
+        threading.Thread(
+            target=cancel_query_in_separate_thread, args=(query_id, query_secret)
+        ).start()
+
+        time.sleep(5)
+
+        long_running_ae.poll_for_status()
+        assert long_running_ae.status == AsyncExecutionStatus.CANCELED
+
+    def test_long_ish_query_canary(self, long_ish_ae: AsyncExecution):
+        """This test verifies that on the current endpoint, the LONG_ISH_QUERY requires
+        at least one poll_for_status call before it is finished. If this test fails, it means
+        the SQL warehouse got faster at executing this query and we should increment the value
+        of GT_FIVE_SECONDS_VALUE
+
+        It would be easier to do this if Databricks SQL had a SLEEP() function :/
+        """
+
+        poll_count = 0
+        while long_ish_ae.is_running:
+            time.sleep(1)
+            long_ish_ae.poll_for_status()
+            poll_count += 1
+
+        assert poll_count > 0
+
+    def test_get_async_execution_and_get_results_without_direct_results(
+        self, long_ish_ae: AsyncExecution
+    ):
+        while long_ish_ae.is_running:
+            time.sleep(1)
+            long_ish_ae.poll_for_status()
 
-    def test_staging_operation(self):
-        """We need to test what happens with a staging operation since this query won't have a result set
-        that user needs. It could be sufficient to notify users that they shouldn't use this API for staging/volumes
-        queries...
+        result = long_ish_ae.get_result().fetchone()
+        assert len(result) == 1
+
+    def test_get_async_execution_with_bogus_query_id(self):
+
+        with self.connection() as conn:
+            with pytest.raises(AsyncExecutionException, match="Query not found"):
+                ae = conn.get_async_execution("bedc786d-64da-45d4-99da-5d3603525803", "ba469f82-cf3f-454e-b575-f4dcd58dd692")
+
+    def test_get_async_execution_with_badly_formed_query_id(self):
+        with self.connection() as conn:
+            with pytest.raises(ValueError, match="badly formed hexadecimal UUID string"):
+                ae = conn.get_async_execution("foo", "bar")
+
+    def test_serialize(self, long_running_ae: AsyncExecution):
+        serialized = long_running_ae.serialize()
+        query_id, query_secret = serialized.split(":")
+
+        with self.connection() as conn:
+            ae = conn.get_async_execution(query_id, query_secret)
+            assert ae.is_running
+
+    def test_get_async_execution_no_results_when_direct_results_were_sent(self):
+        """It remains to be seen whether results can be fetched repeatedly from a "picked up" execution.
```

Comment:
self nit: will remove this comoment. results _can_ be fetched repeatedly until they've been expired by the thrift server.